rfcs/0000_code_guidance_for_researchers_and_community.md

```diff
@@ -0,0 +1,104 @@
+    - Note that this will not be enshrined in any protection rules, since doing so would enforce a PR process whenever the fork / branches are synced with upstream changes.
+- During your work, ensure you regularly update your fork and its branches with any upstream changes to `thousandbrainsproject/tbp.monty`. This will reduce the chance that a longer RP-project is associated with significant merge conflicts when you later wish to merge it into the primary Monty code-base.
+- When a research prototype has been fully implemented and its merits established through evaluations and visualizations, the `main` branch is ready to form the basis of an Implementation Project.
+- The Implementation Project (IP) involves any necessary refactoring of the Research Prototype to ensure it adheres to the [guidelines of the platform](https://thousandbrainsproject.readme.io/docs/code-style-guide) before merging into `tbp.monty`. For more information, see [RFC 14 Conducting Research While Building a Stable Platform](https://github.com/thousandbrainsproject/tbp.monty/blob/main/rfcs/0014_conducting_research_while_building_a_stable_platform.md).
+- Note that the IP process will often involve a series of atomic PRs as well, rather than a single major PR. However, each one will be constrained to follow the platform guidelines.
+
+### Regarding Code Quality Specified in Other Guideline
+- When writing code for the RP, you may choose to follow the [official guidance for the platform](https://thousandbrainsproject.readme.io/docs/code-style-guide). 
+- The advantage of doing so while working on the RP is that it can reduce the amount of work required for the IP stage.
+- However, adhering to these guidelines is often not possible without first carrying out a significant refactor of the code, or through a time-consuming design process.
+- As such, researchers should not follow these guidelines if it will introduce significant delays. Rather, the guidance they provide (e.g., being mindful of how state is added or accessed in code) can provide general tips for writing high-quality code from the beginning.
+
+### Other Comments
+- Your new fork will not have any controls in place to prevent force pushes from other [Members of the TBP organization](https://github.com/orgs/thousandbrainsproject/people). We therefore ask that everyone exercise care in working with the forks of other team members.
+- Similarly, GitHub Workflows (such as unit tests and linting) will not automatically run in your new fork. It is recommended that you regularly run linting and unit-tests in your local environment when developing. 
+
+### Alternative Workflows
+- We discussed using branches within existing personal forks as a way of managing Research Prototypes. However, this risks becoming overwhelming with the number of features. It also does not provide the same visibility to the community about existing prototypes that are reaching a mature stage of development.
+
```

Comment:
One other alternative that I really liked was to use the `monty_project_template` https://github.com/thousandbrainsproject/tbp.monty_project_template to start a project that just uses Monty as a dependency. I enjoyed that workflow for our ultrasound project as it clearly isolated all the changes made in custom classes and it was clear to see just from the files in the repo what the customizations were (instead of requiring a diff view). It was also easy to pin to monty versions and decide when to upgrade the version and make related changes. Is there anything that speaks against that workflow?

Reply:
Ah sorry, yes definitely worth mentioning it. I've added more discussion of alternative workflows, and tried to be less prescriptive in which one to follow (rather, just highlighting pros and cons).